mongocrypt/src/native.rs

```diff
@@ -0,0 +1,48 @@
+use std::borrow::Borrow;
+
+pub(crate) type Destroy<T> = unsafe extern "C" fn(*mut T);
+
+pub(crate) struct OwnedPtr<T> {
+    inner: *mut T,
+    destroy: Destroy<T>,
+}
+
+#[cfg(all(test, feature = "compile_fail"))]
+#[test]
+fn use_after_free() {
+    unsafe extern "C" fn destroy(_ptr: *mut ()) {}
+    let owned: OwnedPtr<()> = OwnedPtr::new(std::ptr::null_mut(), destroy);
```

Comment:
```suggestion
    let owned: OwnedPtr<()> = OwnedPtr::steal(std::ptr::null_mut(), destroy);
```

Reply:
Whoops, done.  Hard to have compile-fail tests that aren't fragile or prone to false passes, I guess.